operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -200,6 +262,75 @@ private String getLockName(String namespace, String name) {
         return result.future();
     }
 
+
+
+    /**
+     * Updates the Status field of the Kafka CR. It diffs the desired status against the current status and calls
+     * the update only when there is any difference in non-timestamp fields.
+     *
+     * @param desiredStatus The KafkaStatus which should be set
+     *
+     * @return
+     */
+    Future<Void> updateStatus(Reconciliation reconciliation, S desiredStatus, List<Condition> unknownAndDeprecatedConditions) {
+        if (desiredStatus == null)  {
+            log.debug("{}: Desired status is null - status will not be updated", reconciliation);
+            return Future.succeededFuture();
+        }
+
+        String namespace = reconciliation.namespace();
+        String name = reconciliation.name();
+        desiredStatus.addConditions(unknownAndDeprecatedConditions);
+
+        Promise<Void> updateStatusPromise = Promise.promise();
+
+        resourceOperator.getAsync(namespace, name).onComplete(getRes -> {
+            if (getRes.succeeded())    {
```

Comment:
if the get fails, we just hang until timeout as the promise is never addressed. We should use a failure handler (preferably with a compose(successhandler, failhandler)

Reply:
I can rewrite it to `compose(..., ...)`. But I think the current code is correct. It fails the promise at the end of the method.